theory/DD/countpairs_impl.c.src

```diff
@@ -221,6 +221,11 @@ int countpairs_DOUBLE(const int64_t ND1, DOUBLE *X1, DOUBLE *Y1, DOUBLE *Z1,
             fprintf(stderr,"ND2 = %12"PRId64" [xmin,ymin,zmin] = [%lf,%lf,%lf], [xmax,ymax,zmax] = [%lf,%lf,%lf]\n",ND2,xmin,ymin,zmin,xmax,ymax,zmax);
         }
     }
+
+    if(options->boxsize < 0 && options->periodic){
+        fprintf(stderr, "boxsize < 0 not allowed with periodic wrap. Please specify a non-zero boxsize or zero to detect the particle extent.\n");
```

Comment:
I would add a print for the value of boxsize -- something like `boxsize = %g should be >= 0`